src/texas_host/core.clj

```diff
@@ -1,27 +1,27 @@
 (ns texas-host.core
   (:gen-class))
 
-(defn remove-card [card pack]
+(defn remove-card [card deck]
   (remove
     (fn [x] (= x card))
-    pack))
+    deck))
 
-(defn get-card [pack]
-  ( let [card (rand-nth pack)]
-  [card (remove-card card pack)]))
+(defn choose-card [deck]
+  [(rand-nth deck)])
 
-(defn get-cards [amount cards pack]
+(defn deal-cards [amount deck]
   (cond
-    (= amount 0) [cards]
-    :else (let [[new-card new-pack] (get-card pack)]
-                   (get-cards (- amount 1) (conj cards new-card) new-pack))))
-
-(defn deal-cards [amount pack]
-  (get-cards amount [] pack))
+    (= amount 0) [[] deck]
+    :else
+    (let [[cards-before deck-before] (deal-cards (- amount 1) deck)]
+      (let [card-chosen (choose-card deck-before)]
+      (let [deck-after (remove-card card-chosen deck-before)]
+        (println "deck-after " card-chosen deck-after)
```

Comment:
Good question. I guess because my it was defined as such in my original algorithm. I'll change it back.